pkg/server/grpc.go

```diff
@@ -51,9 +52,13 @@ type grpcServer struct {
 func newGRPCServer(config *GRPCConfig, server rekorServer) *grpcServer {
 	var opts []grpc.ServerOption
 
-	opts = append(opts, grpc.ChainUnaryInterceptor(
-		getMetrics().serverMetrics.UnaryServerInterceptor(),
-	))
+	opts = append(opts,
+		grpc.ChainUnaryInterceptor(getMetrics().serverMetrics.UnaryServerInterceptor()),
+		grpc.ConnectionTimeout(config.timeout),
+		grpc.KeepaliveParams(keepalive.ServerParameters{MaxConnectionIdle: config.timeout}),
+		// explicitly set to 4MB. https://github.com/grpc/grpc-go/blob/cdbdb759dd67c89544f9081f854c284493b5461c/server.go#L59C39-L59C54
+		grpc.MaxRecvMsgSize(defaultMaxSizeBytes),
```

Comment:
can this be a flag value, `max-request-body-size`, and passed through configuration?

Reply:
There is also the DefaultMaxHeaderBytes, which by default is 1MB, but some providers like GCP load balancer limit it to 8KB. Should we also make that configurable? 

- https://pkg.go.dev/net/http#pkg-constants
- https://cloud.google.com/load-balancing/docs/https/custom-headers#:~:text=the%20same%20name.-,Limitations,KB%20or%2016%20request%20headers.

Reply:
That default seems fine and not something we'd need to modify.